nemo/collections/nlp/data/machine_translation/preproc_mt_data.py

```diff
@@ -0,0 +1,406 @@
+                        decoder_tokenizer=self.decoder_tokenizer,
+                        max_seq_length=cfg.train_ds.max_seq_length,
+                        tokens_in_batch=cfg.train_ds.tokens_in_batch,
+                        lines_per_dataset_fragment=cfg.train_ds.get('lines_per_dataset_fragment'),
+                        num_batches_per_tarfile=cfg.train_ds.get('num_batches_per_tarfile'),
+                        min_seq_length=1,
+                    )
+                    # update config
+                    self._cfg.train_ds.tar_files = self.tar_files_to_string(self.train_tar_files)
+                    self._cfg.train_ds.metadata_file = self.train_metadata_file
+                    logging.info(
+                        f"Using tarred dataset created at {self._cfg.train_ds.tar_files} and metadata created at {self._cfg.train_ds.metadata_file}"
+                    )
+                else:
+                    self.train_tar_files = cfg.train_ds.get('tar_files')
+                    self.train_metadata_file = cfg.train_ds.get('metadata_file')
+                    logging.info(
+                        f"Using tarred dataset from config at {self.train_tar_files} and metadata from {self.train_metadata_file}"
+                    )
+
+    def tar_files_to_string(self, tar_files):
+        """ Tar files are generated in the following format: basename.number.tar
+            Where number is an integer from 1 to the number of tar files.
+            We convert this list to a string that can be used in the model config to specify
+            tarred datasets: basename_OP_1..num_tar_files_CL_.tar 
+
+        Args:
+            tar_files (List[str]): List of tar files generated by preprocess_dataset
+        """
+        num_tar_files = len(tar_files)
+        split_on_dot = tar_files[0].split('.')
+        basename = '.'.join(split_on_dot[0:-2])
+        tar_file_string = f'{basename}._OP_1..{num_tar_files}_CL_.tar'
+        return tar_file_string
+
+    def get_enc_dec_tokenizers(
+        self,
+        encoder_tokenizer_name=None,
+        encoder_tokenizer_model=None,
+        encoder_bpe_dropout=0.0,
+        decoder_tokenizer_name=None,
+        decoder_tokenizer_model=None,
+        decoder_bpe_dropout=0.0,
+    ):
+
+        if encoder_tokenizer_name != 'yttm' or decoder_tokenizer_name != 'yttm':
+            raise NotImplementedError(f"Currently we only support yttm tokenizer.")
+
+        encoder_tokenizer = get_tokenizer(
+            tokenizer_name=encoder_tokenizer_name,
+            tokenizer_model=encoder_tokenizer_model,
+            bpe_dropout=encoder_bpe_dropout,
+        )
+        decoder_tokenizer = get_tokenizer(
+            tokenizer_name=decoder_tokenizer_name,
+            tokenizer_model=decoder_tokenizer_model,
+            bpe_dropout=decoder_bpe_dropout,
+        )
+
+        return encoder_tokenizer, decoder_tokenizer
+
+    # TODO: add local or global rank 0 decorator
```

Comment:
If this is done, wouldn't it mean that we don't need to pass trainer instance to MTDataPreproc constructor?

Reply:
That's true, Trainer instance is only used to get rank information.